backend/src/api/boards.ts

```diff
@@ -52,4 +62,39 @@ router.get("/users/:id", async (req, res) => {
   res.status(200).json(boards);
 });
 
+router.delete("/:id", async (req, res) => {
+  const { id } = req.params;
+
+  const deletedBoard = await dalBoard.remove(id)
+  if (deletedBoard) {
+    const deletedPosts = await dalPost.removeByBoard(id)
+    const deletedLikes = await dalLike.removeByBoard(id)
+    const deletedComments = await dalComment.removeByBoard(id)
+    const deletedBuckets = await dalBucket.removeByBoard(id)
+    const deletedNotifications = await dalNotification.removeByBoard(id)
+    const deletedTags = await dalTag.removeByBoard(id)
+    const deletedTrace = await dalTrace.removeByBoard(id)
+    const deletedWorkflow = await dalWorkflow.removeByBoard(id)
+    const deletedProjectBoard = await dalProject.removeBoard(
+      deletedBoard!.projectID,
```

Comment:
Also, when you move all of this logic into one `dalBoard` function, create a session transaction so the deletes are consistent. Example here: https://github.com/encorelab/ck-board/blob/develop/backend/src/repository/dalPost.ts#L59